frontend/src/components/modals/settings/SettingsForm.tsx

```diff
@@ -43,40 +39,31 @@ function SettingsForm({
 }: SettingsFormProps) {
   const { t } = useTranslation();
   const { isOpen: isVisible, onOpenChange: onVisibleChange } = useDisclosure();
-  const [isAgentSelectEnabled, setIsAgentSelectEnabled] = React.useState(false);
-  const [usingCustomModel, setUsingCustomModel] = React.useState(
-    settings.USING_CUSTOM_MODEL,
```

Comment:
This state was handled to keep track whether user decided to use a custom model or not, to display on the UI. If you set a custom model now through the advanced settings, save, and reopen the modal, it may be confusing at first glance why the model/provider fields are now empty

Reply:
Ah shoot, I meant to switch into advanced mode if the selected model isn't found in the list! I can open a follow-up for that behavior